simple_history/management/commands/clean_duplicate_history.py

```diff
@@ -94,9 +104,45 @@ def _process(self, to_process, date_back=None, dry_run=True):
                 )
 
             for o in model_query.iterator():
-                self._process_instance(o, model, stop_date=stop_date, dry_run=dry_run)
+                self._process_instance(o, model, stop_date=stop_date, dry_run=dry_run, batch_mode=batch_mode, deletion_list=deletion_list)
+
+            if batch_mode and not dry_run:
+                batch_count = (len(deletion_list)) // batch_size
+                if (len(deletion_list)) % batch_size > 0:
+                    batch_count += 1
+
+                self.log(
+                    "Starting deletion with {batch_count} batches of {batch_size} history records of {model}\n".format(batch_count=batch_count, batch_size=batch_size, model=model)
+                )
+
+                for batch_index in range(batch_count):
+                    with transaction.atomic():
```

Comment:
We have a 10k row transaction limit, if you run multiple batches under a transaction it could hit the limit. Each batch-deletion should be independent. Also want to avoid having `time.sleep()` inside a transaction, the longer a transaction runs the more likely it is to fail.

Reply:
Oh my mistake on the batch per transaction part, sorry about that!